scanner/src/main/kotlin/Scanner.kt

```diff
@@ -208,42 +205,25 @@ class Scanner(
             }
         }
 
-        val scanResults = if (scannerConfig.includeFilesWithoutFindings) {
-            filteredScanResults.mapTo(mutableSetOf()) { scanResult ->
-                val allPaths = controller.getAllFileLists()[scanResult.provenance]?.files?.mapTo(mutableSetOf()) {
-                    it.path
-                }.orEmpty()
-
-                val pathsWithFindings = scanResult.summary.licenseFindings.mapTo(mutableSetOf()) { it.location.path }
-                val pathsWithoutFindings = allPaths - pathsWithFindings
-
-                val findingsThatAreNone = pathsWithoutFindings.mapTo(mutableSetOf()) {
-                    LicenseFinding(SpdxConstants.NONE, TextLocation(it, TextLocation.UNKNOWN_LINE))
-                }
-
-                scanResult.copy(
-                    summary = scanResult.summary.copy(
-                        licenseFindings = scanResult.summary.licenseFindings + findingsThatAreNone
-                    )
-                )
-            }
-        } else {
-            filteredScanResults
-        }
-
         val scannerIds = scannerWrappers.mapTo(mutableSetOf()) { it.descriptor.id }
         val scanners = packages.associateBy({ it.id }) { scannerIds }
 
         val issues = controller.getIssues()
 
-        return ScannerRun.EMPTY.copy(
+        val scannerRun = ScannerRun.EMPTY.copy(
             config = scannerConfig,
             provenances = provenances,
             scanResults = scanResults,
             files = files,
             scanners = scanners,
             issues = issues
         )
+
+        if (scannerConfig.includeFilesWithoutFindings) {
+            return scannerRun.padNoneLicenseFindings()
+        }
+
+        return scannerRun
```

Comment:
Nit: More Kotlin-ish would be to extract the `return` like

```kotlin
return if (scannerConfig.includeFilesWithoutFindings) {
    scannerRun.padNoneLicenseFindings()
} else {
    scannerRun
}
```

or even

```kotlin
return scannerRun.takeUnless { scannerConfig.includeFilesWithoutFindings }
    ?: scannerRun.padNoneLicenseFindings()
```

Reply:
Resolved in 16172627c47bd9d19c5c617726ce1a61591cbd4e